### PR TITLE
Monitor req sizes

### PR DESCRIPTION
**Pull request by jalvz**

### CODE REVIEW

File: decoder/decoder.go [hunk 1]
```diff
@@ -42,21 +65,33 @@ func readRequestJSONData(maxSize int64) Reader {
...
 			var err error
 			reader, err = gzip.NewReader(reader)
 			if err != nil {
 				return nil, err
 			}
 		}
+		if knownCLen {
+			uncompressedLengthAcc.Add(cLen)
+		}
+		io.Pipe()
```
Comment (simitt) [inline decoder/decoder.go]: Is this a leftover as you don't use the returned `PipeReader` and `PipeWriter`

File: decoder/decoder.go [hunk 2]
```diff
@@ -42,21 +65,33 @@ func readRequestJSONData(maxSize int64) Reader {
...
+				gzipLengthAcc.Add(cLen)
+			}
 			var err error
 			reader, err = gzip.NewReader(reader)
 			if err != nil {
 				return nil, err
 			}
 		}
+		if knownCLen {
+			uncompressedLengthAcc.Add(cLen)
```
Comment (simitt) [inline decoder/decoder.go]: What do you think about moving this into the `switch` `default` section to only cover non-compressed requests and not have an overall counter?

File: decoder/decoder.go [hunk 3]
```diff
@@ -13,18 +13,41 @@ import (
...
 
 type Reader func(req *http.Request) (io.ReadCloser, error)
 type Decoder func(req *http.Request) (map[string]interface{}, error)
 
+var (
+	decoderMetrics        = monitoring.Default.NewRegistry("apm-server.decoder", monitoring.PublishExpvar)
+	deflateLengthAcc      = monitoring.NewInt(decoderMetrics, "deflate.content-length")
+	gzipLengthAcc         = monitoring.NewInt(decoderMetrics, "gzip.content-length")
+	uncompressedLengthAcc = monitoring.NewInt(decoderMetrics, "uncompressed.content-length")
+	readerAcc             = monitoring.NewInt(decoderMetrics, "reader.size")
```
Comment (simitt) [inline decoder/decoder.go]: What does `Acc` stand for here?

File: decoder/decoder.go (hunk 1 above)
Comment (jalvz) [reply decoder/decoder.go]: yes, is a leftover

File: decoder/decoder.go (hunk 2 above)
Comment (jalvz) [reply decoder/decoder.go]: there isn't an overall counter, all the switch cases return from the function. but i can put this a a default.

File: decoder/decoder.go (hunk 3 above)
Comment (jalvz) [reply decoder/decoder.go]: accumulator. i am not counting with `Inc()`, but accumulating up large numbers, i want to make explicit how these monitors are used.

File: decoder/decoder.go (hunk 2 above)
Comment (jalvz) [reply decoder/decoder.go]: actually you are right. i don't know what i am saying.

**Pull request merged**